programs/svm-spoke/src/instructions/refund_claims.rs

```diff
@@ -57,22 +58,27 @@ pub struct ClaimRelayerRefund<'info> {
     #[account(mint::token_program = token_program)]
     pub mint: InterfaceAccount<'info, Mint>,
 
-    // Token address has been checked when executing the relayer refund leaf and it is part of claim account derivation.
-    #[account(mut, token::mint = mint, token::token_program = token_program)]
+    // TODO: In EVM we currently allow claiming refunds to any address, but probably should restrict it there as well.
```

Comment:
well I think that's a feature, no? an alternative implementation that we could structure here, given we are using addresses (owner) rather than token_account (ATA/TA) is to simply require that the `signer` is the set `refund_address` and then let the caller send the funds to any recipient TA they want.

The reason it was done in the EVM is that if you set the refund address to a blocked address then you're assets are stuck forever unless you can send them somewhere else. Under the assumption that blacklists are implemented universally and consistently, this feature does not let you bi-pass blacklists or anything like that as you would not be able to deposit in the first place. rather, it provides a clean recovery flow, assuming you can prove ownership of the recipient address.

point being: we should be consistent here and could consider doing the same change to simply verify signer and then let you choose which TA to send things to.

Reply:
If claim can be made by anyone on behalf of the relayer then this still better be canonical ATA as otherwise it would be cumbersome for the relayer to reconcile their refunds.

Though, we could create an additional flavor to this method where only the relayer can call it and claim to any token account they control?

Reply:
ye, the latter is what I'm suggesting. only the relayer can claim it, assuming they are the signer which matches with the original repayment recipient in the original fill. this tracks with the EVM implementation more directly: https://github.com/across-protocol/contracts/pull/672/files#r1827795557

Reply:
but this does relate to the point in the other comment thread about _enabling others to execute the claim on behalf of the relayer if the transfer is possible_. the additional flavour might then be the best of both worlds:
- anyone can execute the claim on behalf of a relayer wherein they create the ATA and we send the tokens to the ATA for the claim. the claim is then tracking the outstanding amount for the address (the owner of the ATA).
- if the above fails due to blacklist, then the owner of the address (signer in this extra flavour) can call to send the tokens to a new ATA. clearly, only if they can prove they are the address (singer) can they do this.

This way, the first flow we can use to auto re-fund relayers if we need to use claim accounts and if relayers get stuck they can still recover their funds on theirown